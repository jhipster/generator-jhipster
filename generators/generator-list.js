/**
 * Copyright 2013-2022 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const Generators = {
  GENERATOR_ADD: 'add',
  GENERATOR_APP: 'app',
  GENERATOR_AWS: 'aws',
  GENERATOR_AZURE_APP_SERVICE: 'azure-app-service',
  GENERATOR_AZURE_SPRING_CLOUD: 'azure-spring-cloud',
  GENERATOR_BASE: 'base',
  GENERATOR_BOOTSTRAP: 'bootstrap',
  GENERATOR_BOOTSTRAP_APPLICATION: 'bootstrap-application',
  GENERATOR_CI_CD: 'ci-cd',
  /* @deprecated */
  GENERATOR_CICD: 'ci-cd',
  GENERATOR_CLIENT: 'client',
  GENERATOR_CLOUDFOUNDRY: 'cloudfoundry',
  GENERATOR_COMMON: 'common',
  GENERATOR_CYPRESS: 'cypress',
  GENERATOR_DATABASE_CHANGELOG: 'database-changelog',
  GENERATOR_DATABASE_CHANGELOG_LIQUIBASE: 'database-changelog-liquibase',
  GENERATOR_DOCKER_COMPOSE: 'docker-compose',
  GENERATOR_ENTITIES: 'entities',
  GENERATOR_ENTITIES_CLIENT: 'entities-client',
  GENERATOR_ENTITY: 'entity',
  GENERATOR_ENTITY_CLIENT: 'entity-client',
  GENERATOR_ENTITY_I_18_N: 'entity-i18n',
  /* @deprecated */
  GENERATOR_ENTITY_I18N: 'entity-i18n',
  GENERATOR_ENTITY_SERVER: 'entity-server',
  GENERATOR_EXPORT_JDL: 'export-jdl',
  GENERATOR_GAE: 'gae',
  GENERATOR_GENERATE_BLUEPRINT: 'generate-blueprint',
  GENERATOR_GRADLE: 'gradle',
  GENERATOR_HEROKU: 'heroku',
  GENERATOR_INFO: 'info',
  GENERATOR_INIT: 'init',
  GENERATOR_JAVA: 'java',
  GENERATOR_KUBERNETES: 'kubernetes',
  GENERATOR_KUBERNETES_HELM: 'kubernetes-helm',
  GENERATOR_KUBERNETES_KNATIVE: 'kubernetes-knative',
  GENERATOR_LANGUAGES: 'languages',
  GENERATOR_MAVEN: 'maven',
  GENERATOR_OPENAPI_CLIENT: 'openapi-client',
  GENERATOR_OPENSHIFT: 'openshift',
  GENERATOR_PAGE: 'page',
  GENERATOR_PROJECT_NAME: 'project-name',
  GENERATOR_SERVER: 'server',
  GENERATOR_SPRING_BOOT: 'spring-boot',
  GENERATOR_SPRING_CONTROLLER: 'spring-controller',
  GENERATOR_SPRING_SERVICE: 'spring-service',
  GENERATOR_UPGRADE: 'upgrade',
  GENERATOR_UPGRADE_CONFIG: 'upgrade-config',
  GENERATOR_WORKSPACES: 'workspaces',
};

module.exports = Generators;
