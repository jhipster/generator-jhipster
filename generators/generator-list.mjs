/**
 * Copyright 2013-2023 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project; see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License; Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing; software
 * distributed under the License is distributed on an "AS IS" BASIS;
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND; either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const GENERATOR_ANGULAR = 'angular';
export const GENERATOR_APP = 'app';
export const GENERATOR_AWS = 'aws';
export const GENERATOR_AZURE_APP_SERVICE = 'azure-app-service';
export const GENERATOR_AZURE_SPRING_CLOUD = 'azure-spring-cloud';
export const GENERATOR_BASE = 'base';
export const GENERATOR_BASE_APPLICATION = 'base-application';
export const GENERATOR_BASE_DOCKER = 'base-docker';
export const GENERATOR_BOOTSTRAP = 'bootstrap';
export const GENERATOR_BOOTSTRAP_APPLICATION = 'bootstrap-application';
export const GENERATOR_BOOTSTRAP_APPLICATION_BASE = 'bootstrap-application-base';
export const GENERATOR_BOOTSTRAP_APPLICATION_CLIENT = 'bootstrap-application-client';
export const GENERATOR_BOOTSTRAP_APPLICATION_SERVER = 'bootstrap-application-server';
export const GENERATOR_CASSANDRA = 'cassandra';
export const GENERATOR_CI_CD = 'ci-cd';
export const GENERATOR_CLIENT = 'client';
export const GENERATOR_CLOUDFOUNDRY = 'cloudfoundry';
export const GENERATOR_COMMON = 'common';
export const GENERATOR_COUCHBASE = 'couchbase';
export const GENERATOR_CYPRESS = 'cypress';
export const GENERATOR_DOCKER = 'docker';
export const GENERATOR_DOCKER_COMPOSE = 'docker-compose';
export const GENERATOR_ELASTICSEARCH = 'elasticsearch';
export const GENERATOR_ENTITIES = 'entities';
export const GENERATOR_ENTITY = 'entity';
export const GENERATOR_EXPORT_JDL = 'export-jdl';
export const GENERATOR_GAE = 'gae';
export const GENERATOR_GENERATE_BLUEPRINT = 'generate-blueprint';
export const GENERATOR_GIT = 'git';
export const GENERATOR_GRADLE = 'gradle';
export const GENERATOR_HEROKU = 'heroku';
export const GENERATOR_INFO = 'info';
export const GENERATOR_INIT = 'init';
export const GENERATOR_KAFKA = 'kafka';
export const GENERATOR_KUBERNETES = 'kubernetes';
export const GENERATOR_KUBERNETES_HELM = 'kubernetes-helm';
export const GENERATOR_KUBERNETES_KNATIVE = 'kubernetes-knative';
export const GENERATOR_LANGUAGES = 'languages';
export const GENERATOR_LIQUIBASE = 'liquibase';
export const GENERATOR_LIQUIBASE_CHANGELOGS = 'liquibase-changelogs';
export const GENERATOR_MAVEN = 'maven';
export const GENERATOR_MONGODB = 'mongodb';
export const GENERATOR_OPENAPI_CLIENT = 'openapi-client';
export const GENERATOR_OPENSHIFT = 'openshift';
export const GENERATOR_PAGE = 'page';
export const GENERATOR_PROJECT_NAME = 'project-name';
export const GENERATOR_REACT = 'react';
export const GENERATOR_SERVER = 'server';
export const GENERATOR_SPRING_CONTROLLER = 'spring-controller';
export const GENERATOR_SPRING_SERVICE = 'spring-service';
export const GENERATOR_UPGRADE = 'upgrade';
export const GENERATOR_VUE = 'vue';
export const GENERATOR_WORKSPACES = 'workspaces';
