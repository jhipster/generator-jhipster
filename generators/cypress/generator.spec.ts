/**
 * Copyright 2013-2025 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { basename, dirname } from 'path';
import { fileURLToPath } from 'url';
import { before, describe, expect, it } from 'esmocha';
import { snakeCase } from 'lodash-es';
import { clientFrameworkTypes, testFrameworkTypes } from '../../lib/jhipster/index.js';
import { AuthenticationTypeMatrix, extendMatrix, fromMatrix, defaultHelpers as helpers, runResult } from '../../lib/testing/index.js';
import { checkEnforcements, shouldSupportFeatures, testBlueprintSupport } from '../../test/support/index.js';
import { GENERATOR_CYPRESS } from '../generator-list.js';
import Generator from './generator.js';

const { CYPRESS } = testFrameworkTypes;
const { ANGULAR, REACT, VUE } = clientFrameworkTypes;

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const generator = basename(__dirname);

const e2eMatrix = extendMatrix(
  fromMatrix({
    ...AuthenticationTypeMatrix,
    cypressAudit: [false, true],
  }),
  {
    clientFramework: [ANGULAR, REACT, VUE],
    withAdminUi: [false, true],
    cypressCoverage: [false, true],
    clientRootDir: [undefined, { value: 'clientRoot/' }, { value: '' }],
  },
);

const e2eSamples = Object.fromEntries(
  Object.entries(e2eMatrix).map(([name, sample]) => [
    name,
    {
      ...sample,
      testFrameworks: [CYPRESS],
    },
  ]),
);
const entities = [
  {
    name: 'EntityA',
    changelogDate: '20220129025419',
  },
];

describe(`generator - ${generator}`, () => {
  it('generator-list constant matches folder name', async () => {
    await expect((await import('../generator-list.js'))[`GENERATOR_${snakeCase(generator).toUpperCase()}`]).toBe(generator);
  });
  shouldSupportFeatures(Generator);
  describe('blueprint support', () => testBlueprintSupport(generator));
  checkEnforcements({ client: true }, GENERATOR_CYPRESS);

  it('samples matrix should match snapshot', () => {
    expect(e2eSamples).toMatchSnapshot();
  });

  Object.entries(e2eSamples).forEach(([name, sampleConfig]) => {
    describe(name, () => {
      before(async () => {
        await helpers.runJHipster(generator).withJHipsterConfig(sampleConfig, entities);
      });

      it('should match generated files snapshot', () => {
        expect(runResult.getStateSnapshot()).toMatchSnapshot();
      });

      it('contains cypress testFramework', () => {
        runResult.assertJsonFileContent('.yo-rc.json', { 'generator-jhipster': { testFrameworks: [CYPRESS] } });
      });

      describe('withAdminUi', () => {
        const { applicationType, withAdminUi, clientRootDir = '' } = sampleConfig;
        const generateAdminUi = applicationType !== 'microservice' && withAdminUi;

        if (applicationType !== 'microservice') {
          const adminUiRoutingTitle = generateAdminUi ? 'should generate admin routing' : 'should not generate admin routing';
          const cypressAdminRoot = clientRootDir ? `${clientRootDir}test/` : 'src/test/javascript/';
          it(adminUiRoutingTitle, () => {
            const assertion = (...args) =>
              generateAdminUi ? (runResult.assertFileContent as any)(...args) : (runResult.assertNoFileContent as any)(...args);

            assertion(
              `${cypressAdminRoot}cypress/e2e/administration/administration.cy.ts`,
              '  metricsPageHeadingSelector,\n' +
                '  healthPageHeadingSelector,\n' +
                '  logsPageHeadingSelector,\n' +
                '  configurationPageHeadingSelector,',
            );

            assertion(
              `${cypressAdminRoot}cypress/e2e/administration/administration.cy.ts`,
              "  describe('/metrics', () => {\n" +
                "    it('should load the page', () => {\n" +
                "      cy.clickOnAdminMenuItem('metrics');\n" +
                "      cy.get(metricsPageHeadingSelector).should('be.visible');\n" +
                '    });\n' +
                '  });\n' +
                '\n' +
                "  describe('/health', () => {\n" +
                "    it('should load the page', () => {\n" +
                "      cy.clickOnAdminMenuItem('health');\n" +
                "      cy.get(healthPageHeadingSelector).should('be.visible');\n" +
                '    });\n' +
                '  });\n' +
                '\n' +
                "  describe('/logs', () => {\n" +
                "    it('should load the page', () => {\n" +
                "      cy.clickOnAdminMenuItem('logs');\n" +
                "      cy.get(logsPageHeadingSelector).should('be.visible');\n" +
                '    });\n' +
                '  });\n' +
                '\n' +
                "  describe('/configuration', () => {\n" +
                "    it('should load the page', () => {\n" +
                "      cy.clickOnAdminMenuItem('configuration');\n" +
                "      cy.get(configurationPageHeadingSelector).should('be.visible');\n" +
                '    });\n' +
                '  });',
            );

            assertion(
              `${cypressAdminRoot}cypress/support/commands.ts`,
              'export const metricsPageHeadingSelector = \'[data-cy="metricsPageHeading"]\';\n' +
                'export const healthPageHeadingSelector = \'[data-cy="healthPageHeading"]\';\n' +
                'export const logsPageHeadingSelector = \'[data-cy="logsPageHeading"]\';\n' +
                'export const configurationPageHeadingSelector = \'[data-cy="configurationPageHeading"]\';',
            );
          });
        }
      });
    });
  });
});
