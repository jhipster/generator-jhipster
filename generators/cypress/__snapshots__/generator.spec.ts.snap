// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator - cypress jwt-cypressAudit(false)-angular-withAdminUi(false)-cypressCoverage(false)-clientRoot/ should match generated files snapshot 1`] = `
{
  ".jhipster/EntityA.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "clientRoot/cypress.config.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/package.json": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/.eslintrc.json": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/account/login-page.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/account/password-page.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/account/register-page.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/account/reset-password-page.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/account/settings-page.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/administration/administration.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/entity/authority.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/entity/entity-a.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/fixtures/integration-test.png": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/plugins/index.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/account.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/commands.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/entity.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/index.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/management.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/navbar.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/tsconfig.json": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - cypress jwt-cypressAudit(true)-vue-withAdminUi(true)-cypressCoverage(true)-clientRoot/ should match generated files snapshot 1`] = `
{
  ".jhipster/EntityA.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "clientRoot/cypress-audits.config.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/cypress.config.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/package.json": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/.eslintrc.json": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/account/login-page.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/account/password-page.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/account/register-page.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/account/reset-password-page.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/account/settings-page.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/administration/administration.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/entity/entity-a.cy.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/e2e/lighthouse.audits.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/fixtures/integration-test.png": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/plugins/global.d.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/plugins/index.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/account.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/commands.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/entity.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/index.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/management.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/support/navbar.ts": {
    "stateCleared": "modified",
  },
  "clientRoot/src/test/javascript/cypress/tsconfig.json": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - cypress oauth2-cypressAudit(false)-angular-withAdminUi(false)-cypressCoverage(false) should match generated files snapshot 1`] = `
{
  ".jhipster/EntityA.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "cypress.config.ts": {
    "stateCleared": "modified",
  },
  "package.json": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/.eslintrc.json": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/administration/administration.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/entity/entity-a.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/fixtures/integration-test.png": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/plugins/index.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/commands.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/entity.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/index.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/management.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/navbar.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/oauth2.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/tsconfig.json": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - cypress oauth2-cypressAudit(true)-vue-withAdminUi(true)-cypressCoverage(true)- should match generated files snapshot 1`] = `
{
  ".jhipster/EntityA.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "cypress-audits.config.ts": {
    "stateCleared": "modified",
  },
  "cypress.config.ts": {
    "stateCleared": "modified",
  },
  "package.json": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/.eslintrc.json": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/administration/administration.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/entity/entity-a.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/lighthouse.audits.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/fixtures/integration-test.png": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/plugins/global.d.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/plugins/index.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/commands.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/entity.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/index.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/management.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/navbar.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/oauth2.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/tsconfig.json": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - cypress samples matrix should match snapshot 1`] = `
{
  "jwt-cypressAudit(false)-angular-withAdminUi(false)-cypressCoverage(false)-clientRoot/": {
    "authenticationType": "jwt",
    "clientFramework": "angular",
    "clientRootDir": "clientRoot/",
    "cypressAudit": false,
    "cypressCoverage": false,
    "testFrameworks": [
      "cypress",
    ],
    "withAdminUi": false,
  },
  "jwt-cypressAudit(true)-vue-withAdminUi(true)-cypressCoverage(true)-clientRoot/": {
    "authenticationType": "jwt",
    "clientFramework": "vue",
    "clientRootDir": "clientRoot/",
    "cypressAudit": true,
    "cypressCoverage": true,
    "testFrameworks": [
      "cypress",
    ],
    "withAdminUi": true,
  },
  "oauth2-cypressAudit(false)-angular-withAdminUi(false)-cypressCoverage(false)": {
    "authenticationType": "oauth2",
    "clientFramework": "angular",
    "clientRootDir": undefined,
    "cypressAudit": false,
    "cypressCoverage": false,
    "testFrameworks": [
      "cypress",
    ],
    "withAdminUi": false,
  },
  "oauth2-cypressAudit(true)-vue-withAdminUi(true)-cypressCoverage(true)-": {
    "authenticationType": "oauth2",
    "clientFramework": "vue",
    "clientRootDir": "",
    "cypressAudit": true,
    "cypressCoverage": true,
    "testFrameworks": [
      "cypress",
    ],
    "withAdminUi": true,
  },
  "session-cypressAudit(false)-angular-withAdminUi(false)-cypressCoverage(false)-": {
    "authenticationType": "session",
    "clientFramework": "angular",
    "clientRootDir": "",
    "cypressAudit": false,
    "cypressCoverage": false,
    "testFrameworks": [
      "cypress",
    ],
    "withAdminUi": false,
  },
  "session-cypressAudit(true)-vue-withAdminUi(true)-cypressCoverage(true)": {
    "authenticationType": "session",
    "clientFramework": "vue",
    "clientRootDir": undefined,
    "cypressAudit": true,
    "cypressCoverage": true,
    "testFrameworks": [
      "cypress",
    ],
    "withAdminUi": true,
  },
}
`;

exports[`generator - cypress session-cypressAudit(false)-angular-withAdminUi(false)-cypressCoverage(false)- should match generated files snapshot 1`] = `
{
  ".jhipster/EntityA.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "cypress.config.ts": {
    "stateCleared": "modified",
  },
  "package.json": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/.eslintrc.json": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/account/login-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/account/password-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/account/register-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/account/reset-password-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/account/settings-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/administration/administration.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/entity/authority.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/entity/entity-a.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/fixtures/integration-test.png": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/plugins/index.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/account.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/commands.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/entity.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/index.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/management.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/navbar.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/tsconfig.json": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - cypress session-cypressAudit(true)-vue-withAdminUi(true)-cypressCoverage(true) should match generated files snapshot 1`] = `
{
  ".jhipster/EntityA.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "cypress-audits.config.ts": {
    "stateCleared": "modified",
  },
  "cypress.config.ts": {
    "stateCleared": "modified",
  },
  "package.json": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/.eslintrc.json": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/account/login-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/account/password-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/account/register-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/account/reset-password-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/account/settings-page.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/administration/administration.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/entity/entity-a.cy.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/e2e/lighthouse.audits.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/fixtures/integration-test.png": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/plugins/global.d.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/plugins/index.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/account.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/commands.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/entity.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/index.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/management.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/support/navbar.ts": {
    "stateCleared": "modified",
  },
  "src/test/javascript/cypress/tsconfig.json": {
    "stateCleared": "modified",
  },
}
`;
